### PR TITLE
Optimize mpt node syncing

### PR DESCRIPTION
**Pull request by peterlimg**
## Fixes

When MPT grows large, the previous way of going through the whole MPT db to find missing nodes would not able to work anymore. So this PR will refactor to remove the whole MPT iteration, and sync all missing nodes in given path only. 

## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- 0chain common: https://github.com/0chain/common/pull/5

### CODE REVIEW

File: code/go/0chain.net/chaincore/chain/worker.go [hunk 1]
```diff
@@ -485,103 +498,117 @@ func (c *Chain) SyncLFBStateWorker(ctx context.Context) {
...
 	}
 
-	Logger.Info("Sync round state, found missing nodes",
-		zap.Int64("round", round),
-		zap.Int("missing_node_num", len(keys)))
+	return keys
+}
 
-	c.GetStateNodes(ctx, keys)
+func (c *Chain) syncRoundStateToStateDB(ctx context.Context, round int64, rootStateHash util.Key) {
```
Comment (dabasov) [inline code/go/0chain.net/chaincore/chain/worker.go]: can we delete this func completely?

File: code/go/0chain.net/miner/protocol_block.go [hunk 2]
```diff
@@ -80,10 +80,12 @@ func (mc *Chain) createFeeTxn(b *block.Block) *transaction.Transaction {
 	return feeTxn
 }
 
-func (mc *Chain) getCurrentSelfNonce(minerId datastore.Key, bState util.MerklePatriciaTrieI) (int64, error) {
+func (mc *Chain) getCurrentSelfNonce(round int64, minerId datastore.Key, bState util.MerklePatriciaTrieI) (int64, error) {
 	s, err := mc.GetStateById(bState, minerId)
 	if err != nil {
 		if err != util.ErrValueNotPresent {
+			mc.SyncMissingNodes(round, util.Path(minerId))
```
Comment (dabasov) [inline code/go/0chain.net/miner/protocol_block.go:87]: should we check ErrNotFound directly here or do we assume that only 2 types of errors can be return?

Comment (peterlimg) [reply code/go/0chain.net/miner/protocol_block.go:87]: Yes, should check ErrNotFound directly, could have other errors

**Review by dabasov — APPROVED** [review]
lgtm

**Pull request merged**